pkg/inject/sidecar.go

```diff
@@ -123,7 +134,22 @@ func getJaeger(name string, jaegers *v1.JaegerList) *v1.Jaeger {
 	return nil
 }
 
-func container(jaeger *v1.Jaeger, dep *appsv1.Deployment) corev1.Container {
+func needUpdate(container *corev1.Container, args []string) bool {
+	if len(args) != len(container.Args) {
+		return true
+	}
+
+	// Assumed sorted flags.
+	for i, flag := range container.Args {
```

Comment:
Do not assume the flags are sorted. See #334. Take a look at what we do in `pkg/deployment`

https://github.com/jaegertracing/jaeger-operator/blob/d271a4480599713d8dfd2de0b2088fe8a114d6ff/pkg/deployment/all_in_one.go#L59-L61

Reply:
The reason I assumed sorted arguments is because in that line the arguments are sorted before the comparison. Anyway, this is not necessary if we decided to take the first approach of updating without compare flags.